### PR TITLE
Add onError prop

### PR DESCRIPTION
**Pull request by patrickedqvist**
## TL;DR
By providing an onError prop a developer will gain control over what to do with the errors. 

## User case
In my case I bypass the `submit` and calls the validation directly using ref on a click that isn't a child of `Form`. However since I'm not able to get the internal state of `Form` I need some sort of way of accessing the information.

Other cases could be to show the errors in toasts, modals and other components that aren't child of `Form`.

My solution is to implement a `onError` that calls when the runner fails or passes within the `validator` function. Is simple and provides more flexibility.

Please consider and discuss my solution.

### CODE REVIEW

**Comment by zackify** [comment]
Can you show your current code? I'm wondering if this should be handled by the component, or if you should instead use `values` and `onValues` (passing in the values and updating them yourself) and exposing a `validate` function from the component, 
ex `import Form, { validate } from 'react-validify`

Then you could validate the values whenever you want.

**Comment by patrickedqvist** [comment]
I can't show the current code but it's similar to the following

```js

class Example extends Component {

  constructor(props) {
    super(props);
    this.state = {
      values: {}
    }
    this.handleOnValues = this.handleOnValues.bind(this);
    this.onPublishClick = this.onPublishClick.bind(this);
  }


  handleOnValues(values) {
    this.setState(() => ({ values }));
  }

  onPublishClick() {
    // Run the validate function in <Form>
    this.validationForm.validate(() => {});

    // this.validationForm.state.errors <-- This does not work as the state isn't exposed
    if ( !objectIsEmpty(this.validationForm.state.errors) ) {
      myApiCall(this.state.values);
    }

  }


  render() {

    <div className="outer-div">

      <div className="inner-div">
        <Form values={this.state.values} onValues={this.handleOnValues} ref={ node => { this.form = node } }>
          …alot of inputs
        </Form>
      </div>

      // Component is outside of <Form> and cannot be moved inside
      <Component onPublishClick={this.onPublishClick}>

    </div>
  }


}

```

A  validate function could definitely work as long as it returns the errors too.

**Comment by zackify** [comment]
I'm going to add an onErrors prop that will expose them when they happen and you can pass them back in to `errors` then you can do everything like you want. also will refactor the component to have form-with-state as the default and a base component that you pass the props into to keep the code cleaner. currently it handles state or props and is complicated. I'll work on it tomorrow, expect a new release then!

**Comment by zackify** [comment]
Check out feature/v3, src/form.js imports the base component, it is completely props based, so you have to do onErrors and pass them in yourself if you want to use the base component. Should have this done today or tomorrow!

**Comment by zackify** [comment]
`npm install reat-validify@3.0.0-beta.1` Check out https://github.com/navjobs/validify/blob/feature/v3/src/form.js. You can import the base form and add onErrors and onValues to it. I put it as a beta even though all tests now pass since it was such a big change. Let me know how it works for you :)
